app/viewmodels/checkAnswers/SupplementaryUnitSummary.scala

```diff
@@ -17,24 +17,35 @@
 package viewmodels.checkAnswers
 
 import controllers.routes
-import models.{CheckMode, UserAnswers}
+import models.{CheckMode, RecordCategorisations, UserAnswers}
 import pages.SupplementaryUnitPage
 import play.api.i18n.Messages
+import queries.RecordCategorisationsQuery
 import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryListRow
 import viewmodels.govuk.summarylist._
 import viewmodels.implicits._
 
 object SupplementaryUnitSummary {
 
-  def row(answers: UserAnswers, recordId: String)(implicit messages: Messages): Option[SummaryListRow] =
-    answers.get(SupplementaryUnitPage(recordId)).map { answer =>
-      SummaryListRowViewModel(
-        key = "supplementaryUnit.checkYourAnswersLabel",
-        value = ValueViewModel(answer),
-        actions = Seq(
-          ActionItemViewModel("site.change", routes.SupplementaryUnitController.onPageLoad(CheckMode, recordId).url)
-            .withVisuallyHiddenText(messages("supplementaryUnit.change.hidden"))
-        )
-      )
+  def row(answers: UserAnswers, recordId: String)(implicit messages: Messages): Option[SummaryListRow] = {
+    val recordCategorisations = answers.get(RecordCategorisationsQuery).getOrElse(RecordCategorisations(Map.empty))
+    val categorisationInfoOpt = recordCategorisations.records.get(recordId)
+    categorisationInfoOpt match {
+      case Some(categorisationInfo) =>
+        val measurementUnit = categorisationInfo.measurementUnit
+        answers.get(SupplementaryUnitPage(recordId)).map { answer =>
+          val value = if (measurementUnit.nonEmpty) s"$answer ${measurementUnit.get.trim}" else answer
```

Comment:
Do we want a summary row if the value is empty? I think this should work (SUDO CODE):


```

(for {
recordCategorisations  <- answers.get(RecordCategorisationsQuery)
categorisationInfo <- recordCategorisations.records.get(recordId)
measurementUnit <- categorisationInfo.measurementUnit
supplementaryUnit <- answers.get(SupplementaryUnitPage(recordId))
} yield supplementaryUnit).map{value => 
SummaryListRowViewModel(
             key = "supplementaryUnit.checkYourAnswersLabel",
             value = ValueViewModel(value),
             actions = Seq(
               ActionItemViewModel("site.change", routes.SupplementaryUnitController.onPageLoad(CheckMode, recordId).url)
                 .withVisuallyHiddenText(messages("supplementaryUnit.change.hidden"))
             )
           )}
```

Reply:
have updated to sue for comprehension now